core/genesis.go

```diff
@@ -282,25 +282,22 @@ func (g *Genesis) ToBlock(db ethdb.Database) *types.Block {
 			primeStatedb.SetState(internal, key, value)
 		}
 	}
-	primeRoot := primeStatedb.IntermediateRoot(false)
 	head := types.EmptyHeader()
 	head.SetNonce(types.EncodeNonce(g.Nonce))
 	head.SetTime(g.Timestamp)
 	head.SetExtra(g.ExtraData)
-	head.SetRoot(primeRoot, common.PRIME_CTX)
-	head.SetRoot(types.EmptyRootHash, common.REGION_CTX) // Not genesis allocs allowed
-	head.SetRoot(types.EmptyRootHash, common.ZONE_CTX)   // Not genesis allocs allowed
+	head.SetRoot(types.EmptyRootHash) // Not genesis allocs allowed
```

Comment:
This is not correct. Genesis allocs are allowed. See how the primeRoot was computed before

Reply:
changed it to apply the genesis allocation